Explorer/Assets/Scripts/ECS/Unity/PrimitiveRenderer/Systems/InstantiatePrimitiveRenderingSystem.cs

```diff
@@ -0,0 +1,107 @@
+            meshPoolRegistry = poolsRegistry.GetReferenceTypePool<Mesh>();
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateNonExistingRendererQuery(World);
+            TrySetupExistingRendererQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBMeshRenderer), typeof(SDKTransform), typeof(TransformComponent))]
+        [None(typeof(PrimitiveRendererComponent))]
+        private void InstantiateNonExistingRenderer(in Entity entity, ref PBMeshRenderer sdkComponent, ref TransformComponent transform)
+        {
+            var rendererComponent = new PrimitiveRendererComponent();
+            var meshComponent = new PrimitiveMeshComponent();
+            Instantiate(setupMeshCases[sdkComponent.MeshCase], ref rendererComponent, ref meshComponent, sdkComponent, ref transform);
+            World.Add(entity, rendererComponent, meshComponent);
+        }
+
+        [Query]
+        [All(typeof(PBMeshRenderer), typeof(SDKTransform), typeof(TransformComponent),
+            typeof(PrimitiveRendererComponent), typeof(PrimitiveMeshComponent))]
+        private void TrySetupExistingRenderer(
+            ref PrimitiveRendererComponent primitiveRendererComponent,
+            ref PrimitiveMeshComponent meshComponent,
+            ref PBMeshRenderer sdkComponent,
+            ref TransformComponent transform)
+        {
+            if (!sdkComponent.IsDirty) return;
+
+            ISetupMesh setupCollider = setupMeshCases[sdkComponent.MeshCase];
+
+            // The model has changed entirely, so we need to reinstall the renderer
+            if (ReferenceEquals(meshComponent.Mesh, null))
+                Instantiate(setupCollider, ref primitiveRendererComponent, ref meshComponent, sdkComponent, ref transform);
+
+            sdkComponent.IsDirty = false;
+        }
+
+
+        /// <summary>
+        ///     It is either called when there is no collider or collider was invalidated before (set to null)
+        /// </summary>
+        private void Instantiate(ISetupMesh meshSetup, ref PrimitiveRendererComponent rendererComponent, ref PrimitiveMeshComponent meshComponent,
+            PBMeshRenderer sdkComponent, ref TransformComponent transformComponent)
+        {
+            MeshRenderer meshRendererGo = rendererPoolRegistry.Get();
+            meshRendererGo.material = new Material(Shader.Find("Universal Render Pipeline/Lit"));
+
+            Mesh mesh = meshPoolRegistry.Get();
+
+            meshSetup.Execute(sdkComponent, mesh);
+
+            meshComponent.Mesh = mesh;
+            meshComponent.SDKType = sdkComponent.MeshCase;
+
+            rendererComponent.MeshRenderer = meshRendererGo;
+            rendererComponent.MeshFilter = meshRendererGo.GetComponent<MeshFilter>();
+            rendererComponent.MeshFilter.sharedMesh = mesh;
+
+            Transform rendererTransform = meshRendererGo.transform;
+            rendererTransform.SetParent(transformComponent.Transform, false);
```

Comment:
Let's introduce extensions for component parenting: it's the same for `Collider` and `Renderer` and will be repeated in the future systems